pkg/virt-controller/watch/migration_test.go

```diff
@@ -1586,6 +1597,124 @@ var _ = Describe("Migration watcher", func() {
 		})
 
 	})
+
+	Context("Migration monitoring", func() {
+		var (
+			vmi           *virtv1.VirtualMachineInstance
+			migration     *virtv1.VirtualMachineInstanceMigration
+			targetPod     *k8sv1.Pod
+			attachmentPod *k8sv1.Pod
+			labels        prometheus.Labels
+		)
+
+		BeforeEach(func() {
+			vmi = newVirtualMachineWithHotplugVolume("testvmi-monitoring-"+rand.String(16), virtv1.Running)
+			migration = newMigration("testmigration", vmi.Name, virtv1.MigrationPending)
+			targetPod = newTargetPodForVirtualMachine(vmi, migration, k8sv1.PodRunning)
+			attachmentPod = newAttachmentPodForVirtualMachine(targetPod, migration, k8sv1.PodRunning)
+
+			vmi.Status.NodeName = "node02"
+			targetPod.Spec.NodeName = "node01"
+			labels = prometheus.Labels{"vmi": vmi.Name, "source": vmi.Status.NodeName, "target": targetPod.Spec.NodeName}
+		})
+
+		It("should update metrics on pending migration", func() {
+			pending := prometheustestutil.ToFloat64(migrations.CurrentPendingMigrations.With(labels))
+
+			migration.Status.Phase = virtv1.MigrationPhaseUnset
+			addMigration(migration)
+			addVirtualMachineInstance(vmi)
+			podFeeder.Add(targetPod)
+
+			shouldExpectMigrationPendingState(migration)
+			controller.Execute()
+
+			Expect(prometheustestutil.ToFloat64(migrations.CurrentPendingMigrations.With(labels))).Should(Equal(pending + 1))
+		})
+
+		It("should update metrics on scheduling migration", func() {
+			scheduling := prometheustestutil.ToFloat64(migrations.CurrentSchedulingMigrations.With(labels))
+
+			addMigration(migration)
+			addVirtualMachineInstance(vmi)
+			podFeeder.Add(targetPod)
+			podFeeder.Add(attachmentPod)
+
+			shouldExpectMigrationSchedulingState(migration)
+			controller.Execute()
+
+			Expect(prometheustestutil.ToFloat64(migrations.CurrentSchedulingMigrations.With(labels))).Should(Equal(scheduling + 1))
+		})
+
+		It("should update metrics on running migration", func() {
+			running := prometheustestutil.ToFloat64(migrations.CurrentRunningMigrations.With(labels))
+
+			vmi.Status.MigrationState = &virtv1.VirtualMachineInstanceMigrationState{
+				MigrationUID:      migration.UID,
+				TargetNode:        "node01",
+				SourceNode:        "node02",
+				TargetNodeAddress: "10.10.10.10:1234",
+				StartTimestamp:    now(),
+			}
+			migration.Status.Phase = virtv1.MigrationTargetReady
+			addMigration(migration)
+			addVirtualMachineInstance(vmi)
+			podFeeder.Add(targetPod)
+			podFeeder.Add(attachmentPod)
```

Comment:
This part is also duplicated in all tests. Maybe this kind of function can make things more clear:
```go
addObjects := func(addAttachmentpod bool) {
	addMigration(migration)
	addVirtualMachineInstance(vmi)
	podFeeder.Add(targetPod)
	if (addAttachmentpod) {
		podFeeder.Add(attachmentPod)
	}
}
```

Reply:
updated